File: 1300/1372B/main.cpp
#include <bits/stdc++.h>
using namespace std;

long long gcd(long long a, long long b) 
{
    if (a == 0) 
        return b;

    return gcd(b % a, a);
}

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long n;
        cin >> n;

        if ((n & 1) == 0)
            cout << n / 2 << " " << n / 2 << "\n";
        else 
        {
            long long a = -1;
            long long b = -1;
            long long lcm = LLONG_MAX;

            for (long long i = 1; i <= sqrt(n); i++) 
            {
                if (n % i == 0) 
                {
                    long long f1 = i;
                    long long l = (f1 * 1LL * (n - f1)) / gcd(f1, n - f1);
                    if (l < lcm) 
                    {
                        lcm = l;
                        a = f1;
                        b = n - f1;
                    }

                    if (i == 1) 
                        continue;

                    long long f2 = n / i;
                    l = (f2 * 1LL * (n - f2)) / gcd(f2, n - f2);
                    if (l < lcm) 
                    {
                        lcm = l;
                        a = f2;
                        b = n - f2;
                    }
                }
            }

            cout << a << " " << b << "\n";
        }
    }
    
    return 0;
}

File: 1300/1372B/main.java
import java.util.*;
public class main
{
    public static long gcd(long a, long b)
    {
        if(a == 0)  
            return b;
        
        return gcd(b%a, a);
    }
    
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long n = sc.nextInt();
            
            if((n&1) == 0)
                System.out.println(n/2 + " " + n/2);
            else
            {
                long a = -1;
                long b = -1;
                
                long lcm = Long.MAX_VALUE;
                
                for(int i=1;i<=Math.sqrt(n);i++)
                {
                    if(n%i == 0)
                    {
                        long f1 = i;
                        long l = ((f1*1L*(n-f1))/gcd(f1, n-f1));
                        if(l < lcm)
                        {
                            lcm = l;
                            a = f1;
                            b = n-f1;
                        }
                        
                        if(i == 1)
                            continue;
                            
                        long f2 = n/i;
                        l = ((f2*1L*(n-f2))/gcd(f2, n-f2));
                        if(l < lcm)
                        {
                            lcm = l;
                            a = f2;
                            b = n-f2;
                        } 
                    }
                }
                
                System.out.println(a + " " + b);
            }
        }
    }
}
